tools/ci/cibuild.sh

```diff
@@ -55,50 +55,135 @@ function add_path {
   EXTRA_PATH=$1:${EXTRA_PATH}
 }
 
-function python-tools {
-  # Python User Env
-  PIP_USER=yes
-  export PIP_USER
-  PYTHONUSERBASE=${prebuilt}/pylocal
-  export PYTHONUSERBASE
-  add_path "${PYTHONUSERBASE}"/bin
-  pip3 install pexpect==4.8.0
-  pip3 install pytest==6.2.5
-  pip3 install pytest-repeat==0.9.1
-  pip3 install pytest-json==0.4.0
-  pip3 install pytest-ordering==0.6
-  pip3 install pyserial==3.5
-  pip3 install pyelftools
-  pip3 install cxxfilt
+function arm-clang-toolchain {
+  add_path "${prebuilt}"/clang-arm-none-eabi/bin
 
-  # MCUboot's tool for image signing and key management
-  if ! command -v imgtool &> /dev/null; then
-    pip3 install imgtool
+  if [ ! -f "${prebuilt}/clang-arm-none-eabi/bin/clang" ]; then
+    cd "${prebuilt}"
+    curl -O -L -s https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/releases/download/release-14.0.0/LLVMEmbeddedToolchainForArm-14.0.0-linux.tar.gz
+    tar zxf LLVMEmbeddedToolchainForArm-14.0.0-linux.tar.gz
+    mv LLVMEmbeddedToolchainForArm-14.0.0 clang-arm-none-eabi
+    rm LLVMEmbeddedToolchainForArm-14.0.0-linux.tar.gz
   fi
+  clang --version
 }
 
-function codechecker {
-  pip3 install CodeChecker
+function arm-gcc-toolchain {
+  add_path "${prebuilt}"/gcc-arm-none-eabi/bin
+
+  if [ ! -f "${prebuilt}/gcc-arm-none-eabi/bin/arm-none-eabi-gcc" ]; then
+    local flavor
+    case ${os} in
+      Darwin)
+        flavor=-darwin
+        ;;
+      Linux)
+        flavor=
+        ;;
+    esac
+    cd "${prebuilt}"
+    curl -O -L -s https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi.tar.xz
+    xz -d arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi.tar.xz
+    tar xf arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi.tar
+    mv arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi gcc-arm-none-eabi
+    patch -p0 < ${nuttx}/tools/ci/patch/arm-none-eabi-workaround-for-newlib-version-break.patch
+    rm arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi.tar
```

Comment:
```suggestion
    mkdir -p "${prebuilt}"/gcc-arm-none-eabi
    curl -SL https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/arm-gnu-toolchain-11.3.rel1${flavor}-x86_64-arm-none-eabi.tar.xz | tar -xzC "${prebuilt}"/gcc-arm-none-eabi --strip 1
    patch -d "${prebuilt}"/gcc-arm-none-eabi -p0 < ${nuttx}/tools/ci/patch/arm-none-eabi-workaround-for-newlib-version-break.patch
```
Please try if this will work

Reply:
sorry. here will not work since I just noticed that file is `xz` and not `gz`

Reply:
BTW, I can't download that file with `curl`on my laptop while all is ok with `wget`

Reply:
Done.